ztp/source-cluster-crs/cluster-crs.yaml

```diff
@@ -0,0 +1,144 @@
+---
+apiVersion: agent-install.openshift.io/v1beta1
+kind: NMStateConfig
+metadata:
+  name: siteconfig.Spec.Clusters.ClusterName
+  namespace: siteconfig.Spec.Clusters.ClusterName
+  labels:
+    nmstate-label: siteconfig.Spec.Clusters.ClusterName
+spec:
+  config: siteconfig.Spec.Clusters.Nodes.NodeNetwork.Config
+  interfaces: siteconfig.Spec.Clusters.Nodes.NodeNetwork.Interfaces
+---
+apiVersion: agent.open-cluster-management.io/v1
+kind: KlusterletAddonConfig
+metadata:
+  name: siteconfig.Spec.Clusters.ClusterName
+  namespace: siteconfig.Spec.Clusters.ClusterName
+spec:
+  clusterName: siteconfig.Spec.Clusters.ClusterName
+  clusterNamespace: siteconfig.Spec.Clusters.ClusterName
+  clusterLabels:
+    cloud: auto-detect
+    vendor: auto-detect
+  applicationManager:
+    enabled: true
+  certPolicyController:
+    enabled: false
+  iamPolicyController:
+    enabled: false
+  policyController:
+    enabled: true
+  searchCollector:
+    enabled: false
+---
+apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+  name: siteconfig.Spec.Clusters.ClusterName
+  labels: siteconfig.Spec.Clusters.ClusterLabels
+spec:
+  hubAcceptsClient: true
+---
+apiVersion: agent-install.openshift.io/v1beta1
+kind: InfraEnv
+metadata:
+  name: siteconfig.Spec.Clusters.ClusterName
+  namespace: siteconfig.Spec.Clusters.ClusterName
+spec:
+  clusterRef:
+    name: siteconfig.Spec.Clusters.ClusterName
+    namespace: siteconfig.Spec.Clusters.ClusterName
+  sshAuthorizedKey: siteconfig.Spec.SshPublicKey
+  agentLabelSelector:
+    matchLabels:
+      cluster-name: siteconfig.Spec.Clusters.ClusterName
+  pullSecretRef:
+    name: siteconfig.Spec.PullSecretRef
+  ignitionConfigOverride: siteconfig.Spec.Clusters.IgnitionConfigOverride
+  nmStateConfigLabelSelector:
+    matchLabels:
+      nmstate-label: siteconfig.Spec.Clusters.ClusterName
+  additionalNTPSources: siteconfig.Spec.Clusters.AdditionalNTPSources
+---
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: siteconfig.Spec.Clusters.ClusterName
+  namespace: siteconfig.Spec.Clusters.ClusterName
+  annotations:
+    inspect.metal3.io: disabled
+    bmac.agent-install.openshift.io/hostname: siteconfig.Spec.Clusters.Nodes.HostName
+  labels:
+    infraenvs.agent-install.openshift.io: siteconfig.Spec.Clusters.ClusterName
+spec:
+  bootMode: siteconfig.Spec.Clusters.Nodes.BootMode
+  bmc:
+    address: siteconfig.Spec.Clusters.Nodes.BmcAddress
+    disableCertificateVerification: true
+    credentialsName: siteconfig.Spec.Clusters.Nodes.BmcCredentialsName.Name
+  bootMACAddress: siteconfig.Spec.Clusters.Nodes.BootMACAddress
+  automatedCleaningMode: disabled
+  online: true
+  rootDeviceHints: siteconfig.Spec.Clusters.Nodes.RootDeviceHints
+  userData: siteconfig.Spec.Clusters.Nodes.UserData
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: siteconfig.Spec.Clusters.ClusterName
+  namespace: siteconfig.Spec.Clusters.ClusterName
+data:
```

Comment:
What is this configmap? Is it the BMH credentials? If so I think the name would be siteconfig.Spec.Clusters.Nodes.BmcCredentialsName.Name

Reply:
this configMap for the extra manifest

Reply:
okay